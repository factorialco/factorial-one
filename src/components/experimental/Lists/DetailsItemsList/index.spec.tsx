import { render } from "@testing-library/react-native";
import React from "react";
import { DetailsItemsList, DetailsItemsListProps } from ".";

jest.mock("expo-clipboard", () => ({
  setString: jest.fn(),
  getString: jest.fn(),
}));

describe("DetailsItemsList", () => {
  it("Snapshot ", () => {
    const content: DetailsItemsListProps = {
      title: "Details",
      details: [
        {
          title: "Legal entity",
          content: {
            type: "item",
            text: "Everyday Software SL",
            action: {
              type: "copy",
            },
          },
        },
        {
          title: "Manager",
          content: {
            type: "person",
            firstName: "Saul",
            lastName: "Dominguez",
            avatarUrl: "https://github.com/sauldom102.png",
            action: {
              type: "generic",
              handlePress: () => console.log("Its work"),
            },
          },
        },
        {
          title: "Teams",
          content: [
            {
              type: "team",
              name: "Management",
              action: {
                type: "generic",
                handlePress: () => console.log("Its work"),
              },
            },
            {
              type: "team",
              name: "Engineering",
              action: {
                type: "generic",
                handlePress: () => console.log("Its work"),
              },
            },
          ],
        },
        {
          title: "Type",
          content: {
            type: "dot-tag",
            text: "Holidays",
            customColor: "#07A2AD",
          },
        },
      ],
    };

    const { toJSON } = render(<DetailsItemsList {...content} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
