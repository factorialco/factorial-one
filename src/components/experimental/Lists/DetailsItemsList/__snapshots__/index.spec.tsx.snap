// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsItemsList Snapshot  1`] = `
<View
  className="flex gap-4"
>
  <Text
    className="mb-1 pl-1.5 text-sm font-semibold text-f1-foreground-secondary"
  >
    DETAILS
  </Text>
  <View
    className="flex gap-3"
  >
    <View
      className="flex gap-0.5"
    >
      <View
        className="min-w-32 max-w-72"
      >
        <Text
          className="px-1.5 text-f1-foreground-secondary mb-0.5"
        >
          Legal entity
        </Text>
        <View
          className="flex flex-shrink justify-center gap-0.5"
        >
          <View
            className="flex rounded font-medium text-f1-foreground *:flex-1"
          >
            <View
              accessibilityLabel="Copy Everyday Software SL"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              className="group flex flex-row justify-between gap-1.5 rounded p-1.5 transition-colors duration-300 active:bg-f1-background-secondary-hover"
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                className="flex flex-row items-center gap-1.5"
              >
                <Text
                  className="line-clamp-5 text-left text-f1-foreground"
                >
                  Everyday Software SL
                </Text>
              </View>
              <View
                className="flex"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  aria-hidden={true}
                  bbHeight="100%"
                  bbWidth="100%"
                  className="shrink-0 w-5 h-5 stroke-md col-start-1 col-end-2 row-start-1 row-end-2 text-f1-icon-bold"
                  fill="none"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M9 20h1.4c3.36 0 5.04 0 6.324-.654a6 6 0 0 0 2.622-2.622C20 15.44 20 13.76 20 10.4V9"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeLinecap",
                        ]
                      }
                      stroke={
                        {
                          "type": 2,
                        }
                      }
                      strokeLinecap={1}
                    />
                    <RNSVGPath
                      d="M8 5h6a3 3 0 0 1 3 3v6a3 3 0 0 1-3 3H8a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "type": 2,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      className="flex gap-0.5"
    >
      <View
        className="min-w-32 max-w-72"
      >
        <Text
          className="px-1.5 text-f1-foreground-secondary mb-0.5"
        >
          Manager
        </Text>
        <View
          className="flex flex-shrink justify-center gap-0.5"
        >
          <View
            className="flex rounded font-medium text-f1-foreground *:flex-1"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              className="group justify-between rounded text-f1-foreground no-underline hover:bg-f1-background-hover focus-visible:outline focus-visible:outline-2 focus-visible:outline-f1-border-selected-bold active:bg-f1-background-secondary-hover flex flex-row items-center gap-1.5 p-1.5"
              collapsable={false}
              focusable={true}
              handlePress={[Function]}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              type="generic"
            >
              <View
                className="flex flex-row items-center gap-1.5"
              >
                <View
                  className="flex inline-flex "
                >
                  <View
                    className="h-fit w-fit"
                  >
                    <View
                      aria-hidden={true}
                      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-5 h-5 rounded-full dark:bg-f1-background-inverse-secondary bg-f1-background"
                      data-a11y-color-contrast-ignore={true}
                      role="img"
                    >
                      <View
                        className="aspect-square h-full w-full"
                      >
                        <Image
                          aria-label="S"
                          source={
                            {
                              "uri": "https://github.com/sauldom102.png",
                            }
                          }
                          style={
                            {
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <Text
                  className="line-clamp-5 text-left text-f1-foreground"
                >
                  Saul Dominguez
                </Text>
              </View>
              <RNSVGSvgView
                align="xMidYMid"
                aria-hidden={true}
                bbHeight="100%"
                bbWidth="100%"
                className="shrink-0 w-5 h-5 stroke-md text-f1-foreground"
                fill="none"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="m9 6 6 6M9 18l6-6"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                    strokeLinecap={1}
                    strokeLinejoin={1}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      className="flex gap-0.5"
    >
      <View
        className="min-w-32 max-w-72"
      >
        <Text
          className="px-1.5 text-f1-foreground-secondary mb-0.5"
        >
          Teams
        </Text>
        <View
          className="flex flex-shrink justify-center gap-0.5"
        >
          <View
            className="flex rounded font-medium text-f1-foreground *:flex-1"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              className="group justify-between rounded text-f1-foreground no-underline hover:bg-f1-background-hover focus-visible:outline focus-visible:outline-2 focus-visible:outline-f1-border-selected-bold active:bg-f1-background-secondary-hover flex flex-row items-center gap-1.5 p-1.5"
              collapsable={false}
              focusable={true}
              handlePress={[Function]}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              type="generic"
            >
              <View
                className="flex flex-row items-center gap-1.5"
              >
                <View
                  className="flex inline-flex "
                >
                  <View
                    className="h-fit w-fit"
                  >
                    <View
                      aria-hidden={true}
                      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-5 h-5 rounded-xs bg-[hsl(theme(colors.yellow.50))]"
                      data-a11y-color-contrast-ignore={true}
                      role="img"
                    >
                      <Text
                        className="text-f1-foreground-inverse/90 text-sm"
                        data-a11y-color-contrast-ignore={true}
                      >
                        M
                      </Text>
                    </View>
                  </View>
                </View>
                <Text
                  className="line-clamp-5 text-left text-f1-foreground"
                >
                  Management
                </Text>
              </View>
              <RNSVGSvgView
                align="xMidYMid"
                aria-hidden={true}
                bbHeight="100%"
                bbWidth="100%"
                className="shrink-0 w-5 h-5 stroke-md text-f1-foreground"
                fill="none"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="m9 6 6 6M9 18l6-6"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                    strokeLinecap={1}
                    strokeLinejoin={1}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            className="flex rounded font-medium text-f1-foreground *:flex-1"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              className="group justify-between rounded text-f1-foreground no-underline hover:bg-f1-background-hover focus-visible:outline focus-visible:outline-2 focus-visible:outline-f1-border-selected-bold active:bg-f1-background-secondary-hover flex flex-row items-center gap-1.5 p-1.5"
              collapsable={false}
              focusable={true}
              handlePress={[Function]}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              type="generic"
            >
              <View
                className="flex flex-row items-center gap-1.5"
              >
                <View
                  className="flex inline-flex "
                >
                  <View
                    className="h-fit w-fit"
                  >
                    <View
                      aria-hidden={true}
                      className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-5 h-5 rounded-xs bg-[hsl(theme(colors.malibu.50))]"
                      data-a11y-color-contrast-ignore={true}
                      role="img"
                    >
                      <Text
                        className="text-f1-foreground-inverse/90 text-sm"
                        data-a11y-color-contrast-ignore={true}
                      >
                        E
                      </Text>
                    </View>
                  </View>
                </View>
                <Text
                  className="line-clamp-5 text-left text-f1-foreground"
                >
                  Engineering
                </Text>
              </View>
              <RNSVGSvgView
                align="xMidYMid"
                aria-hidden={true}
                bbHeight="100%"
                bbWidth="100%"
                className="shrink-0 w-5 h-5 stroke-md text-f1-foreground"
                fill="none"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="m9 6 6 6M9 18l6-6"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                    strokeLinecap={1}
                    strokeLinejoin={1}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      className="flex gap-0.5"
    >
      <View
        className="min-w-32 max-w-72"
      >
        <Text
          className="px-1.5 text-f1-foreground-secondary mb-0.5"
        >
          Type
        </Text>
        <View
          className="flex flex-shrink justify-center gap-0.5"
        >
          <View
            className="flex items-start pt-1"
          >
            <View
              className="flex items-start"
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                className="flex flex-row items-center justify-start gap-0.5 rounded-full py-0.5 pr-2 pl-1 border border-solid border-f1-border-secondary"
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  aria-hidden={true}
                  className="m-1 aspect-square w-2 rounded-full"
                  style={
                    {
                      "backgroundColor": "#07A2AD",
                    }
                  }
                />
                <Text
                  className="line-clamp-1 text-base font-medium text-f1-foreground"
                >
                  Holidays
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
